sql-db/narayana-transactions/src/main/java/io/quarkus/ts/transactions/recovery/TransactionLogsResource.java

```diff
@@ -31,15 +37,26 @@ public class TransactionLogsResource {
 
     private final DataSource dataSource;
     private final EnumMap<TransactionExecutor, TransactionRecoveryService> typeToSvc;
+    private final DataSource xaDataSource2;
+    /**
+     * For Oracle we use 2 separate databases because there is Oracle-specific optimization in place which detects that
+     * the database is same even though we have 2 different data sources and they recognize that 2-phase commit is not
+     * necessary.
+     */
+    private final boolean isOracleTest;
```

Comment:
This variable only used to check whether we need to apply changes to xaDataSource2. I presume, it can be replaced with null check on xaDataSource2